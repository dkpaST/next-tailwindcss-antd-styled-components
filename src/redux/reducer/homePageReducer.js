import { HOME_PAGE_TITLE } from '../actions/homePageAction';

export const initialState = {
  title: 'Hello Next.js'
};

const homePageReducer = (state = initialState, action) => {
  switch (action.type) {
    case HOME_PAGE_TITLE:
      return {
        ...state,
        title: action.payload.newTitle
      };
    default:
      return state;
  }
};

export default homePageReducer;
