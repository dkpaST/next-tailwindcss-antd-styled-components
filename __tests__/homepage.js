/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
const mockStore = configureStore([]);
import Index from '../pages/index';

import { initialState } from '../redux/reducer/homePageReducer';


it('My Connected React-Redux Component', () => {
  let store;
  let tree;
  store = mockStore({ "homePage": initialState });
  tree = renderer.create(
    <Provider store={store}>
      <Index />
    </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});