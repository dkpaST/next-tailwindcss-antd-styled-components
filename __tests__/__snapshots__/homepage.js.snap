// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div
  className="jsx-1276654382 container"
>
  <main
    className="jsx-1276654382"
  >
    <h1
      className="jsx-1276654382 title bg-red-600"
    >
      Welcome to 
      <a
        className="jsx-1276654382"
        href="https://nextjs.org"
      >
        Next.js!
      </a>
    </h1>
    <p
      className="jsx-1276654382 description"
    >
      Get started by editing 
      <code
        className="jsx-1276654382"
      >
        pages/index.js
      </code>
    </p>
    <div
      className="jsx-1276654382 grid"
    >
      <a
        className="jsx-1276654382 card"
        href="https://nextjs.org/docs"
      >
        <h3
          className="jsx-1276654382"
        >
          Documentation →
        </h3>
        <p
          className="jsx-1276654382"
        >
          Find in-depth information about Next.js features and API.
        </p>
      </a>
      <a
        className="jsx-1276654382 card"
        href="https://nextjs.org/learn"
      >
        <h3
          className="jsx-1276654382"
        >
          Learn →
        </h3>
        <p
          className="jsx-1276654382"
        >
          Learn about Next.js in an interactive course with quizzes!
        </p>
      </a>
      <a
        className="jsx-1276654382 card"
        href="https://github.com/zeit/next.js/tree/master/examples"
      >
        <h3
          className="jsx-1276654382"
        >
          Examples →
        </h3>
        <p
          className="jsx-1276654382"
        >
          Discover and deploy boilerplate example Next.js projects.
        </p>
      </a>
      <a
        className="jsx-1276654382 card"
        href="https://zeit.co/new?filter=next.js&utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
      >
        <h3
          className="jsx-1276654382"
        >
          Deploy →
        </h3>
        <p
          className="jsx-1276654382"
        >
          Instantly deploy your Next.js site to a public URL with ZEIT Now.
        </p>
      </a>
    </div>
  </main>
  <footer
    className="jsx-1276654382"
  >
    <a
      className="jsx-1276654382"
      href="https://zeit.co?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
      rel="noopener noreferrer"
      target="_blank"
    >
      Powered by 
      <img
        alt="ZEIT Logo"
        className="jsx-1276654382"
        src="/zeit.svg"
      />
    </a>
  </footer>
</div>
`;
